src/data_management/rasters/Raster.js

```diff
@@ -13,63 +13,100 @@ import { FormattedMessage } from "react-intl";
-      // checkboxes.
+      // is checked, to be able to find the raster back from the checked checkboxes.
       checkboxes: [],
-      checkboxAllCheckboxesChecked: false,
       searchTerms: ""
     };
     this.handleNewRasterClick = this.handleNewRasterClick.bind(this);
     this.handleDeleteRasterClick = this.handleDeleteRasterClick.bind(this);
     this.checkAllCheckBoxes = this.checkAllCheckBoxes.bind(this);
     this.clickRegularCheckbox = this.clickRegularCheckbox.bind(this);
     this.handleNewRasterClick = this.handleNewRasterClick.bind(this);
-    this.loadRastersOnPage = this.loadRastersOnPage.bind(this);
   }
   componentDidMount() {
     const { page } = this.state;
-    this.loadRastersOnPage(page, this.state.searchTerms);
+    this.getRastersFromApi(page, this.state.searchTerms);
   }
 
-  loadRastersOnPage(page, searchContains) {
-    const url = searchContains
-      ? // ordering is done by filter
-        `/api/v3/rasters/?page=${page}&name__icontains=${searchContains}` // &organisation__unique_id=${organisationId},
-      : // ordering is done so latest rasters first
-        `/api/v3/rasters/?ordering=-last_modified&page=${page}`;
+  filterSortRasters = (rasters, searchContains) => {
+    const filteredRasters = rasters.filter(
+      e =>
+        e.name.toLowerCase().includes(searchContains.toLowerCase()) ||
+        e.description.toLowerCase().includes(searchContains.toLowerCase())
+    );
+    const sortedFilteredRasters = filteredRasters.sort(
+      (a, b) => a.last_modified > b.last_modified
+    );
+    return sortedFilteredRasters;
+  };
+
+  paginateRasters = (rasters, page) => {
+    const paginatedRasters = rasters.slice(
+      this.state.pageSize * (page - 1),
+      this.state.pageSize * page
+    );
+    return paginatedRasters;
+  };
+
+  refreshRasterFilteringAndPaginationAndUpdateState = (
+    rasters,
+    page,
+    searchTerms
+  ) => {
+    const filteredSortedRasters = this.filterSortRasters(rasters, searchTerms);
+    const paginatedRasters = this.paginateRasters(filteredSortedRasters, page);
+    const checkboxes = this.createCheckboxDataFromRaster(paginatedRasters);
+
+    this.setState({
+      isFetching: false,
+      total: filteredSortedRasters.length,
+      rasters: rasters,
+      filteredSortedRasters: filteredSortedRasters,
+      paginatedRasters: paginatedRasters,
+      page: page,
+      checkAllCheckBoxes: false,
+      checkboxes: checkboxes,
+      searchTerms: searchTerms
+    });
+  };
+
+  getRastersFromApi = (page, searchContains) => {
+    // const url = searchContains
+    //   ? // ordering is done by filter
+    //     `/api/v3/rasters/?writable=true&page=${page}&name__icontains=${searchContains}` // &organisation__unique_id=${organisationId},
+    //   : // ordering is done so latest rasters first
+    //     `/api/v3/rasters/?writable=true&ordering=-last_modified&page=${page}`;
+    const url = "/api/v3/rasters/?writable=true&page_size=100000";
```

Comment:
Is het een idee om van de page_size value een constante te maken?
(Mocht het in de toekomst aangepast moeten worden.)